x-pack/plugins/transform/server/routes/api/transforms_nodes.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isPopulatedObject } from '../../../common/utils/object_utils';
+
+import { RouteDependencies } from '../../types';
+
+import { addBasePath } from '../index';
+
+import { wrapError, wrapEsError } from './error_utils';
+
+interface NodesAttributes {
+  attributes: {
+    'transform.node': string;
+  };
+}
+type Nodes = Record<string, NodesAttributes>;
+
+const isNodes = (arg: unknown): arg is Nodes => {
+  return (
+    isPopulatedObject(arg) &&
+    Object.values(arg).every(
+      (node) =>
+        isPopulatedObject(node) &&
+        {}.hasOwnProperty.call(node, 'attributes') &&
+        isPopulatedObject(node.attributes) &&
+        {}.hasOwnProperty.call(node.attributes, 'transform.node')
+    )
+  );
+};
+
+export function registerTransformNodesRoutes({ router, license }: RouteDependencies) {
+  /**
+   * @apiGroup Transform Nodes
+   *
+   * @api {get} /api/transforms/_nodes Transform Nodes
+   * @apiName GetTransformNodes
+   * @apiDescription Get transform nodes
+   */
+  router.get<undefined, undefined, undefined>(
+    {
+      path: addBasePath('transforms/_nodes'),
+      validate: false,
+    },
+    license.guardApiRoute<undefined, undefined, undefined>(async (ctx, req, res) => {
+      try {
+        const {
+          body: { nodes },
+        } = await ctx.core.elasticsearch.client.asInternalUser.nodes.info({
+          filter_path: 'nodes.*.attributes',
+        });
+
+        let count = 0;
+        if (isNodes(nodes)) {
+          for (const { attributes } of Object.values(nodes)) {
+            if (attributes['transform.node'] === 'true') {
```

Comment:
as `isNodes` is checking the structure of this object, could this loop just be something like
```
count = Object.values(nodes.attributes).filter(n => n === true))
```